components/user-header.js

```diff
@@ -71,14 +71,22 @@ function HeaderPhoto ({ user, isMe }) {
             }
           }
         })
+      },
+      onCompleted ({ setPhoto: photoId }) {
+        // add random query param for cache busting
+        const randomParam = (Math.random() + 1).toString(16).substring(2)
+        const src = `https://${process.env.NEXT_PUBLIC_AWS_UPLOAD_BUCKET}.s3.amazonaws.com/${user.photoId}?r=${randomParam}`
+        setSrc(src)
```

Comment:
Is this needed because you are reusing the `photoId`?

Reply:
Yes. Without this, the avatar won't update after upload since the browser won't know that the image has been updated. The random parameter makes the browser fetch the new image. When you revisit the page, the browser seemed to reload the image even without a random parameter. But maybe I didn't test the caching behavior enough since I thought it will load the old image again with the plain URL.

Reply:
Okay. not entirely sure why this works, but it seems to work. The stale image in the cache seems to have been overwritten, even though we used a random param :thinking: 


https://github.com/stackernews/stacker.news/assets/27162016/e6b58bd9-9271-474e-a262-3c90d6ed623d